src/main/java/com/amplitude/HttpTransport.java

```diff
@@ -47,13 +47,14 @@ class HttpTransport {
   private long flushTimeout;
 
   HttpTransport(
-      HttpCall httpCall, AmplitudeCallbacks callbacks, AmplitudeLog logger, long flushTimeout) {
+      HttpCall httpCall, AmplitudeCallbacks callbacks, AmplitudeLog logger,
+      long flushTimeout, ExecutorService sendThreadPool, ExecutorService retryThreadPool) {
     this.httpCall = httpCall;
     this.callbacks = callbacks;
     this.logger = logger;
     this.flushTimeout = flushTimeout;
-    retryThreadPool = Executors.newFixedThreadPool(10);
-    sendThreadPool = Executors.newFixedThreadPool(20);
+    this.retryThreadPool = (retryThreadPool == null) ? Executors.newFixedThreadPool(10) : retryThreadPool;
+    this.sendThreadPool = (sendThreadPool == null) ? Executors.newFixedThreadPool(40) : sendThreadPool;
```

Comment:
I dont follow why we had to increase the count from 20 to 40.

Reply:
bump @izaaz